### PR TITLE
Add command suggestion when misspelled

### PR DESCRIPTION
**Pull request by Miroito**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,10 +25,12 @@
     "node": ">=12"
   },
   "dependencies": {
+    "@types/fuzzyset.js": "^0.0.2",
```
Comment (vostrnad) [inline package.json:28]: @types packages should be in `devDependencies`.

File: src/modules/commands/CommandHandler.ts [hunk 2]
```diff
@@ -236,6 +237,30 @@ export class CommandHandler<T = unknown> {
...
+      const match = fz.get(message.text.slice(prefix.length))
+      if (match === null || match === undefined) {
+        // if match is less than 33% null is returned by default in the module
+        return false
+      }
+
+      const distthreshold = 0.6 // 60% of the keyword/alias has to be right for a suggestion to be relevant
+      if (match[0][0] < 1 - distthreshold) {
+        message.reply(
+          `Unknown command, use **${prefix}help** to get a list of all commands`,
```
Comment (vostrnad) [inline src/modules/commands/CommandHandler.ts:256]: Missing period at the end of a sentence.

File: src/modules/commands/CommandHandler.ts [hunk 3]
```diff
@@ -236,6 +237,30 @@ export class CommandHandler<T = unknown> {
...
+
+      const fz = fuzzySet(commandsAliases)
+      const match = fz.get(message.text.slice(prefix.length))
+      if (match === null || match === undefined) {
+        // if match is less than 33% null is returned by default in the module
+        return false
+      }
+
+      const distthreshold = 0.6 // 60% of the keyword/alias has to be right for a suggestion to be relevant
+      if (match[0][0] < 1 - distthreshold) {
```
Comment (vostrnad) [inline src/modules/commands/CommandHandler.ts:254]: Why the `1 - distthreshold`?

File: src/modules/commands/CommandHandler.ts [hunk 4]
```diff
@@ -236,6 +237,30 @@ export class CommandHandler<T = unknown> {
...
 
+    if (message.text.startsWith(`${prefix}`)) {
+      const commandsAliases: string[] = []
+      this._commands.forEach((command) => {
+        commandsAliases.push(command.keyword, ...command.alias)
+      })
+
+      const fz = fuzzySet(commandsAliases)
+      const match = fz.get(message.text.slice(prefix.length))
+      if (match === null || match === undefined) {
```
Comment (vostrnad) [inline src/modules/commands/CommandHandler.ts:248]: This can be simplified to `if (!match)`.

Comment (Miroito) [reply src/modules/commands/CommandHandler.ts:248]: Noice

File: package.json (hunk 1 above)
Comment (vostrnad) [reply package.json:28]: `yarn remove @types/fuzzyset.js`
`yarn add -D @types/fuzzyset.js`

**Review by vostrnad — APPROVED** [review]

**Pull request merged**